lib/omnibus/packagers/solaris.rb

```diff
@@ -88,17 +111,44 @@ def write_pkginfo_content
         BASEDIR=#{install_dirname}
         PKG=#{project.package_name}
         NAME=#{project.package_name}
-        ARCH=#{`uname -p`.chomp}
+        ARCH=#{safe_architecture}
         VERSION=#{pkgmk_version}
         CATEGORY=application
         DESC=#{project.description}
         VENDOR=#{project.maintainer}
         EMAIL=#{project.maintainer}
-        PSTAMP=#{`hostname`.chomp + Time.now.utc.iso8601}
+        PSTAMP=#{hostname}#{Time.now.utc.iso8601}
       EOF
       File.open staging_dir_path('pkginfo'), 'w+' do |f|
         f.write pkginfo_content
       end
     end
+
+    #
+    # Generate the Solaris file using +pkg*+.
+    #
+    # @return [void]
+    #
+    def create_solaris_file
+      shellout! "pkgmk -o -r #{install_dirname} -d #{staging_dir} -f #{staging_dir_path('Prototype')}"
+      shellout! "pkgchk -vd #{staging_dir} #{project.package_name}"
+      shellout! "pkgtrans #{staging_dir} #{package_path} #{project.package_name}"
+    end
+
+    #
+    # The architecture for this Solaris package.
+    #
+    # @return [String]
+    #
+    def safe_architecture
+      # The #intel? and #sparc? helpers come from chef-sugar
+      if intel?
```

Comment:
Because there is a bug in chef-sugar:
https://github.com/sethvargo/chef-sugar/pull/121

Without that fix this spec fails:
https://github.com/chef/omnibus/blob/074002633760d29d86d5c651768493e7c00e666d/spec/unit/packagers/solaris_spec.rb#L225-L231